en/docs/SetupAndInstall/UpgradingWSO2APIManager/upgrading-from-250-to-300.md

```diff
@@ -3,350 +3,1676 @@
+        If you have configured WSO2CONFIG_DB in the previous API-M version, add a new entry to the `<API-M_3.0.0_HOME>/repository/conf/deployment.toml` as below.
 
-        !!! info
-    If you are working with a **clustered/distributed API Manager setup** , follow step [5](#UpgradingfromthePreviousRelease-step5-2.2.0) on the Gateway node.
+        ```
+        [database.config]
+        type = "mysql"
+        url = "jdbc:mysql://localhost:3306/config_db"
+        username = "username"
+        password = "password"
+
+        ```
+        
+4.  Update `<API-M_3.0.0_HOME>/repository/resources/conf/default.json` file by pointing to the **WSO2UM_DB**.
+
+    ```
+    "realm_manager": {
+        "data_source": "WSO2USER_DB",
+        "properties": {
+        "isCascadeDeleteEnabled": true,
+        "initializeNewClaimManager": true
+        }
+    }
+    ```
+
+    !!! info
+        In API-M 3.0.0, you do not need to configure the registry configurations as you did in the `<OLD_API-M_HOME>/repository/conf/registry.xml` file and the user database configurations as you did in in the `<OLD_API-M_HOME>/repository/conf/user-mgt.xml` file, as those configurations have been handled internally.
 
 
-5.  Move all your Synapse configurations to API-M 2.6.0 pack.
+5.  Copy the relevant JDBC driver to the `<API-M_3.0.0_HOME>/repository/components/lib` folder.
+
+6.  Move all your Synapse configurations to API-M 3.0.0 pack.
     -   Move your Synapse super tenant configurations.
-        Copy the contents in the `<CURRENT_API-M_HOME>/repository/deployment/server/synapse-configs/default` directory and replace the contents in the `<API-M_2.6.0_HOME>/repository/deployment/server/synapse-configs/default` directory with the copied contents.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/deployment/server/synapse-configs/default` directory and replace the contents in the `<API-M_3.0.0_HOME>/repository/deployment/server/synapse-configs/default` directory with the copied contents.
     -   Move all your tenant Synapse configurations.
-        Copy the contents in the `<CURRENT_API-M_HOME>/repository/tenants` directory and replace the contents in the `<API-M_2.6.0_HOME>/repository/tenants` directory with the copied contents.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/tenants` directory and replace the contents in the `<API-M_3.0.0_HOME>/repository/tenants` directory with the copied contents.
 
-#### Step 1.2 - Optionally, migrate the configurations for WSO2 API-M Analytics
+    !!! warning
+        When moving the Synapse configurations, **do not replace** the following set of files as they contain some modifications in API-M 3.0.0 version.
 
-!!! warning
-This step is **only required** if you have WSO2 API-M-Analytics configured in your current deployment.
-Follow the steps below to migrate APIM Analytics 2.5.0 to APIM Analytics 2.6.0
+        -   \_RevokeAPI_.xml
+        -   \_cors_request_handler_.xml
+        -   main.xml
```

Comment:
Shall we add an attention note as below just after this warning as in 2.6 to 3.0 doc?
`!!! attention 
        If you are working with a **clustered/distributed API Manager setup**, follow this step on the **Gateway** node.`

Reply:
Added